governance/program/src/state/proposal.rs

```diff
@@ -15,51 +25,273 @@ pub struct Proposal {
-    /// to move instruction to voting state) and bring mint to net 0 tokens outstanding. Each signatory gets 1 (serves as flag)
-    pub signatory_mint: Pubkey,
-
-    /// Admin ownership mint. One token is minted, can be used to grant admin status to a new person
-    pub admin_mint: Pubkey,
-
     /// Indicates which Governing Token is used to vote on the Proposal
     /// Whether the general Community token owners or the Council tokens owners vote on this Proposal
-    pub voting_token_type: GoverningTokenType,
+    pub governing_token_mint: Pubkey,
 
-    /// Current state of the proposal
+    /// Current proposal state
     pub state: ProposalState,
 
-    /// Total signatory tokens minted, for use comparing to supply remaining during draft period
-    pub total_signatory_tokens_minted: u64,
+    /// The TokenOwnerRecord representing the user who created and owns this Proposal
+    pub token_owner_record: Pubkey,
+
+    /// The number of signatories assigned to the Proposal
+    pub signatories_count: u8,
+
+    /// The number of signatories who already signed
+    pub signatories_signed_off_count: u8,
 
     /// Link to proposal's description
     pub description_link: String,
 
     /// Proposal name
     pub name: String,
 
-    /// When the Proposal ended voting - this will also be when the set was defeated or began executing naturally
-    pub voting_ended_at: Option<Slot>,
+    /// When the Proposal was created and entered Draft state
+    pub draft_at: Slot,
+
+    /// When Signatories started signing off the Proposal
+    pub signing_off_at: Option<Slot>,
 
     /// When the Proposal began voting
-    pub voting_began_at: Option<Slot>,
+    pub voting_at: Option<Slot>,
 
-    /// when the Proposal entered draft state
-    pub created_at: Option<Slot>,
+    /// When the Proposal ended voting and entered either Succeeded or Defeated
+    pub voting_completed_at: Option<Slot>,
 
-    /// when the Proposal entered completed state, also when execution ended naturally.
-    pub completed_at: Option<Slot>,
+    /// When the Proposal entered Executing state
+    pub executing_at: Option<Slot>,
 
-    /// when the Proposal entered deleted state
-    pub deleted_at: Option<Slot>,
+    /// When the Proposal entered final state Completed or Cancelled and was closed
+    pub closed_at: Option<Slot>,
 
     /// The number of the instructions already executed
     pub number_of_executed_instructions: u8,
 
     /// The number of instructions included in the proposal
     pub number_of_instructions: u8,
+}
+
+impl AccountMaxSize for Proposal {
+    fn get_max_size(&self) -> Option<usize> {
+        Some(self.name.len() + self.description_link.len() + 163)
+    }
+}
+
+impl IsInitialized for Proposal {
+    fn is_initialized(&self) -> bool {
+        self.account_type == GovernanceAccountType::Proposal
+    }
+}
+
+impl Proposal {
```

Comment:
I replaced the first two  `assert_can_add_signatory()` and `assert_can_remove_signatory()` with a single `assert_can_edit_signatories()` 
I kept the third one `assert_can_sign_off()` but removed the redundant check `self.signatories_count == 0`. I kept it because the name reflects a state transition check (there are going to be more like this) but it also gives a different error which provides a better description for the guarded condition